pkg/endpoint/endpoint.go

```diff
@@ -933,8 +935,8 @@ func (e *Endpoint) removeDirectory() {
 
 func (e *Endpoint) RemoveDirectory() {
 	e.Mutex.Lock()
+	defer e.Mutex.Unlock()
```

Comment:
That is also an artifact of the debugging but my thought was if e.removeDirectory() panics, is there a possibility to leak the lock?